scripts/image-saver.py

```diff
@@ -1,85 +1,179 @@
+        Path where the hdf5 file containing the images should be saved to.
+    filename: string
+        Name of hdf5 file images are saved in. If empty, will default to
+        {camera}-{timestamp}. If not empty, timestamp will be appended
+        to ensure all captures have unique filenames.
     """
-    current_name = args.filename
+    # Default filename is camera-timestamp
+    if not filename:
+        filename = f'{camera.name}-{int(time())}'
+    # Setting up hdf51 plugin
+    camera.hdf51.file_name.put(filename)
+    camera.hdf51.file_template.put('%s%s_%03d.h5')
+    camera.hdf51.num_capture.put(num_images)
+    camera.hdf51.auto_save.put(True)
+    camera.hdf51.file_write_mode.put('Capture')
+    camera.hdf51.enable.put('Enabled')
+    camera.hdf51.capture.put('Capture')
+    # Creating callbacks to know when collecting/saving images is complete
+    collecting = True
+    writing = True
+
+    def _collecting_cb(value, **kwargs):
+        """Get a callback for collecting changes"""
+        nonlocal collecting
+        collecting = value
+
+    def _writing_cb(value, **kwargs):
+        """Get a callback when writing changes"""
+        nonlocal writing
+        writing = value
+    # Start collecting and saving
+    collect_uid = camera.hdf51.capture.subscribe(_collecting_cb)
+    write_uid = camera.hdf51.write_file.subscribe(_writing_cb)
+    # Clean up subscribers and disable plugin when finished
+    while collecting and camera.hdf51.num_captured.get() < num_images:
+        sleep(0.1)
+    camera.hdf51.capture.unsubscribe(collect_uid)
+    while writing:
+        sleep(0.1)
+    camera.hdf51.write_file.unsubscribe(write_uid)
+    camera.hdf51.enable.put('Disabled')
+
+
+def load_camera(hutch, cam_name):
+    """
+    Creates camera object.
+
+    Gets camera details based on name and hutch-specific
+    camviewer config file and uses build_cam to create an
+    individual camera object. Will exit if the config file
+    or the camera cannot be found.
+
+    Parameters
+    ----------
+    hutch: string
+        Name of hutch the camera is in, needed to find and
+        load the correct config file.
+    cam_name: string
+        Name of camera to load.
+
+    Returns
+    -------
+    camera: pcdsdevices.areadetector.detectors.PCDSAreaDetector
+        Camera object created by build_cam with hdf51 plugin
+        member to take and save images.
+    """
+    # Get cam_load functions
+    syspath.append(f'/reg/g/pcds/pyps/apps/hutch-python/{hutch}'
+                   '/dev/hutch-python/hutch_python/cam_load.py')
+    from hutch_python.cam_load import interpret_lines, build_cam
+    # Get config for cam_name
+    try:
+        f = open(f'/reg/g/pcds/pyps/config/{hutch}/camviewer.cfg')
+    except Exception as e:
+        print(e)
+        exit('Could not open camera config file, '
+             f'/reg/g/pcds/pyps/config/{hutch}/camviewer.cfg')
+    cam_line = [line for line in f.readlines() if 'im2k0'.upper() in line]
```

Comment:
I think this line must be in error, there are plenty of cameras that aren't named `im2k0`